src/main/java/roomescape/service/waiting/WaitingQueryService.java

```diff
@@ -0,0 +1,32 @@
+package roomescape.service.waiting;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import roomescape.domain.waiting.Waiting;
+import roomescape.domain.waiting.WaitingWithRank;
+import roomescape.dto.reservation.ReservationAndWaitingResponseDto;
+import roomescape.dto.waiting.WaitingResponseDto;
+import roomescape.repository.JpaWaitingRepository;
+
+import java.util.List;
+
+@Service
+@Transactional(readOnly = true)
+public class WaitingQueryService {
+
+    JpaWaitingRepository waitingRepository;
+
+    public WaitingQueryService(JpaWaitingRepository jpaWaitingRepository) {
+        this.waitingRepository = jpaWaitingRepository;
+    }
+
+    public List<ReservationAndWaitingResponseDto> findMyWaiting(long memberId){
+        List<WaitingWithRank> byMemberId = waitingRepository.findByMemberId(memberId);
+        return byMemberId.stream().map(ReservationAndWaitingResponseDto::new).toList();
+    }
+
+    public List<WaitingResponseDto> findAllWaiting(){
+        List<Waiting> all = waitingRepository.findAll();
+        return all.stream().map(waiting -> WaitingResponseDto.from(waiting)).toList();
```

Comment:
사소한 리뷰) 요렇게도 가능합니다. ㅎㅎ
```suggestion
        return all.stream().map(WaitingResponseDto::from).toList();
```

Reply:
좋습니다! 반영하였습니다.